Check if we have __GLIBC__ defined at all before checking version
http://bugs.alpinelinux.org/issues/3423

This fixes the following when compiled with musl libc:
warning: unix_send_fd: your system has no support for file descriptor passing


--- ./src/util/sys_defs.h.orig
+++ ./src/util/sys_defs.h
@@ -810,7 +810,7 @@
 # define KERNEL_VERSION(a,b,c) (LINUX_VERSION_CODE + 1)
 #endif
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2,2,0)) \
-	|| (__GLIBC__ < 2)
+	|| (defined(__GLIBC__) && (__GLIBC__ < 2))
 # define CANT_USE_SEND_RECV_MSG
 # define DEF_SMTP_CACHE_DEMAND	0
 #else
